### DISCUSSION TITLE
What if the argument channel_slug is None ?

### DISCUSSION THREAD

**Discussion by iamleson98** [saleor/saleor#9366] [Q&A] [answered]
https://github.com/saleor/saleor/blob/f8e2b49703d995d4304d5a690dbe9c83631419d0/saleor/graphql/product/filters.py#L309

Hi @maarcingebala and @IKarbowiak, I mention you two since I see both of you are main contributors of Saleor.

I tried clone the codebases for the backend and dashboard and have tried the product exporting feature, that work consists of multiple functions and the function in the link above is one of them.

I see you pass `channel_slug` argument to this function.
But as I know, this parameter can be of type `None` or `str`. I wonder when it's `None`, will the result still be reliable ?
Could any of you look at the code above and answer my question.

Thank you.

**Comment by IKarbowiak** [comment, marked as answer]
Hi @iamleson98,

When the `channel_slug` is `None`, nothing will be returned. Filtering by stock availability should be always performed for a specified channel.
